boltdesignsystem/bolt
# Twig 2 upgrade (Drupal 9 compatibility)

## Jira

https://pegadigitalit.atlassian.net/browse/DS-489

## Summary

Updates pattern Lab to twig 2.0 for compatibility with Drupal 9

## Details

* Fixes deprecated/improper twig usages in Bolt source code so that it's compatible with Twig 2.
* Patches/hacks pattern lab php for compatibility with twig 2.

## How to test

* On this branch, confirm that anywhere twig is installed in a `vendor` directory (`vendor/twig/twig`), version 2.x is being used.  The quickest way to do this is by looking at `vendor/twig/twig/CHANGELOG` and confirming the latest lines references twig 2.14.
* Confirm no regressions in pattern lab: https://feature-ds-489-twig-2.boltdesignsystem.com/

## Release notes

Support for twig 2 added.
